src/gui/addnewtorrentdialog.cpp

```diff
@@ -186,32 +186,24 @@ void AddNewTorrentDialog::setTopLevel(bool value)
 
 int AddNewTorrentDialog::savePathHistoryLength()
 {
-    return savePathHistoryLengthSetting();
+    const int defaultHistoryLength = 8;
+    return settings()->loadValue(KEY_SAVEPATHHISTORYLENGTH, defaultHistoryLength).toInt();
 }
 
 void AddNewTorrentDialog::setSavePathHistoryLength(int value)
 {
     Q_ASSERT(value >= minPathHistoryLength);
     Q_ASSERT(value <= maxPathHistoryLength);
+
     const int oldValue = savePathHistoryLength();
     if (oldValue != value) {
-        savePathHistoryLengthSetting() = value;
-        settings()->storeValue(KEY_SAVEPATHHISTORY,
-                               QStringList(settings()->loadValue(KEY_SAVEPATHHISTORY).toStringList().mid(0, value)));
+        settings()->storeValue(KEY_SAVEPATHHISTORYLENGTH
+            , std::max(minPathHistoryLength, std::min(maxPathHistoryLength, value)));
```

Comment:
qBound?

Reply:
Besides, it's pointless because of asserts above.

Reply:
I'll just change to `qBound`.

Reply:
Then you need to use it before `if (oldValue != value)` to prevent false positives.
Moreover, assertions above become unnecessary.

Reply:
thanks, fixed.